### PR TITLE
add support for create/join group commands

### PR DESCRIPTION
**Pull request by niteshbalusu11**
Added support for creating or joining groups non-interactively through new commands.

Files changed:
index.js
groups/index.js

New files:
groups/create_group_channel.js
groups/join_group_channel.js
groups/get_join_details.js

### CODE REVIEW

File: groups/create_group_channel.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+      // Get identity public key
+      getIdentity: ['validate', ({}, cbk) => getIdentity({lnd: args.lnd}, cbk)],
+
+      // Get methods to confim partial signing is supported
+      getMethods: ['validate', ({}, cbk) => getMethods({lnd: args.lnd}, cbk)],
+
+      // Make sure that partially signing a PSBT is valid
+      confirmSigner: ['getMethods', ({getMethods}, cbk) => {
+        if (!getMethods.methods.find(n => n.endpoint === signPsbtEndpoint)) {
```
Comment (alexbosworth) [inline groups/create_group_channel.js:106]: There is a method `verifyAccess` for this

File: groups/join_group_channel.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+        return cbk();
+      },
+
+      // Get methods to confim partial signing is supported
+      getMethods: ['validate', ({}, cbk) => getMethods({lnd: args.lnd}, cbk)],
+
+      // Make sure that partially signing a PSBT is valid
+      confirmSigner: ['getMethods', ({getMethods}, cbk) => {
+        if (!getMethods.methods.find(n => n.endpoint === signPsbtEndpoint)) {
```
Comment (alexbosworth) [inline groups/join_group_channel.js:56]: same here, could use verifyAccess

File: groups/get_join_details.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+const tokensAsBigUnit = tokens => (tokens / 1e8).toFixed(8);
+
+/** Ask to confirm joining a group
+
+  {
+    ask: <Ask Function>
+    code: <Group Invite Code String>
+    lnd: <Authenticated LND API Object>
+    logger: <Winston Logger Object>
+    rate: <Max Fee Rate For Open Number>
```
Comment (alexbosworth) [inline groups/get_join_details.js]: this variable doesn't seem to be used

File: groups/get_join_details.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+const isCode = n => !!n && n.length === 98;
+const isPublicKey = n => !!n && /^0[2-3][0-9A-F]{64}$/i.test(n);
+const niceName = n => `${n.alias} ${n.id}`.trim();
+const times = 2 * 60;
+const tokensAsBigUnit = tokens => (tokens / 1e8).toFixed(8);
+
+/** Ask to confirm joining a group
+
+  {
+    ask: <Ask Function>
```
Comment (alexbosworth) [inline groups/get_join_details.js]: this doesn't seem to be part of the method signature

File: groups/join_group_channel.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        return cbk();
+      }],
+
+      // Decode the code and get group details
+      getJoinDetails: ['confirmSigner', ({}, cbk) => {
+        return getJoinDetails({
+          code: args.code, 
+          lnd: args.lnd, 
+          logger: args.logger,
+          rate: args.max_rate
```
Comment (alexbosworth) [inline groups/join_group_channel.js]: not necessary?

File: groups/get_join_details.js (hunk 3 above)
Comment (niteshbalusu11) [reply groups/get_join_details.js]: I logged all the details down below in one place. So pushed it along.

File: groups/join_group_channel.js (hunk 5 above)
Comment (niteshbalusu11) [reply groups/join_group_channel.js]: I didn't understand? That part is used to decode the code and get all the details.

File: groups/create_group_channel.js (hunk 1 above)
Comment (niteshbalusu11) [reply groups/create_group_channel.js:106]: Looks like verify access needs the macaroon to be passed? And also, what's the permissions I need to pass to it?

File: groups/get_join_details.js (hunk 3 above)
Comment (alexbosworth) [reply groups/get_join_details.js]: the logging is actually logging the real fee, not the max fee though

File: groups/join_group_channel.js (hunk 5 above)
Comment (alexbosworth) [reply groups/join_group_channel.js]: I just don't think this variable is ever referenced in that function, and that makes sense, like why would get group details as a function need to know about what the max auto-accepting fee rate would be

File: groups/create_group_channel.js (hunk 1 above)
Comment (alexbosworth) [reply groups/create_group_channel.js:106]: hmm yeah nm that sounds annoying

**Comment by niteshbalusu11** [comment]
Fixed.

**Pull request merged**